### ISSUE TITLE
Error in Style/RedundantBegin with empty begin-end block

### ISSUE THREAD

**Issue by daniel-rikowski** [rubocop/rubocop#9539]
The Style/RedundantBegin cop generates an error on empty begin-end blocks.

**Example:**

```ruby
begin
end
```

results in `undefined method 'rescue_type?' for nil:NilClass`

--------

## Expected behavior

rubocop should not throw an error.

## Actual behavior

rubocop throws the error `undefined method 'rescue_type?' for nil:NilClass`

Relevant call stack:
```
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rubocop-1.10.0/lib/rubocop/cop/style/redundant_begin.rb:105:in `contain_rescue_or_ensure?'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rubocop-1.10.0/lib/rubocop/cop/style/redundant_begin.rb:88:in `on_kwbegin'
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/rubocop-1.10.0/lib/rubocop/cop/commissioner.rb:100:in `public_send'
```

## Steps to reproduce the problem

Run the Style/RedundantBegin cop on a file with an empty begin-end block:

```ruby
begin
end
```

```ruby
begin
  # Comment
end
```

## RuboCop version

```
1.10.0 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 3.0.0 x64-mingw32)
  - rubocop-performance 1.9.2
  - rubocop-rails 2.9.1
  - rubocop-rspec 2.2.0
```
